data_structures/src/chain.rs

```diff
@@ -530,74 +512,23 @@ impl AsRef<Transaction> for Transaction {
 }
 
 /// Input data structure
-#[derive(Debug, Eq, PartialEq, Clone, Serialize, Deserialize, ProtobufConvert)]
+#[derive(Debug, Default, Eq, PartialEq, Clone, Serialize, Deserialize, ProtobufConvert)]
 #[protobuf_convert(pb = "witnet::TransactionBody_Input")]
-pub enum Input {
-    Commit(CommitInput),
-    DataRequest(DataRequestInput),
-    Reveal(RevealInput),
-    ValueTransfer(ValueTransferInput),
+pub struct Input {
+    output_pointer: OutputPointer,
 }
 
 impl Input {
+    /// Create a new Input from an OutputPointer
+    pub fn new(output_pointer: OutputPointer) -> Self {
+        Self { output_pointer }
+    }
     /// Return the [`OutputPointer`](OutputPointer) of an input.
     pub fn output_pointer(&self) -> OutputPointer {
-        // TODO: Potential refactor (redundant `match`)
-        match self {
-            Input::Commit(input) => OutputPointer {
-                transaction_id: input.transaction_id,
-                output_index: input.output_index,
-            },
-            Input::DataRequest(input) => OutputPointer {
-                transaction_id: input.transaction_id,
-                output_index: input.output_index,
-            },
-            Input::Reveal(input) => OutputPointer {
-                transaction_id: input.transaction_id,
-                output_index: input.output_index,
-            },
-            Input::ValueTransfer(input) => OutputPointer {
-                transaction_id: input.transaction_id,
-                output_index: input.output_index,
-            },
-        }
+        self.output_pointer.clone()
     }
 }
 
-/// Value transfer input transaction data structure
-#[derive(Debug, Eq, PartialEq, Clone, Serialize, Deserialize, ProtobufConvert, Default)]
-#[protobuf_convert(pb = "witnet::TransactionBody_Input_ValueTransferInput")]
-pub struct ValueTransferInput {
-    pub transaction_id: Hash,
-    pub output_index: u32,
-}
-
-/// Commit input transaction data structure
-#[derive(Debug, Eq, PartialEq, Clone, Serialize, Deserialize, ProtobufConvert, Default)]
-#[protobuf_convert(pb = "witnet::TransactionBody_Input_CommitInput")]
-pub struct CommitInput {
-    pub transaction_id: Hash,
-    pub output_index: u32,
-    pub nonce: u64,
```

Comment:
Since the `nonce` field was removed from `CommitInput`, it should be added to `RevealOutput`.

Reply:
This is provisional anyway, because as soon as we get VRFs, we may use them for producing the commitments as well. Under that scenario, the right place for the revealed nonce will be the `proof` area of the reveal transaction.